tests/common/test_utils.py

```diff
@@ -430,61 +419,35 @@ def construct_run_config_result(avg_gpu_metric_values,
 
     return run_config_result
 
+class ConfigCommandProfileSubClass(ConfigCommandProfile):
+    # Override ConfigCommandProfile.set_config_values() method
+    def set_config_values(self, args, yaml_config):
+        self._set_field_values(args, yaml_config)
+        self._autofill_values()
+
+class CLISubclass(CLI):
+    # Override CLI.parse() method
+    def parse(self, yaml_config):
+        config_profile = ConfigCommandProfileSubClass()
+        config_profile.set_config_values(argparse.Namespace(), yaml_config)
+        return config_profile
 
-def construct_constraint_manager(constraints=None, yaml_str=None):
+def construct_constraint_manager(yaml_config: Optional[Dict[str, Dict[str, Any]]]=None):
```

Comment:
Can yaml_config be a string for simplicity? Then you can utilize the existing evaluate_mock_config() code, and all tests will follow the same path.

Reply:
This has been fixed 👍